### PR TITLE
Feature/set port range

### PR DESCRIPTION
**Pull request by lbb**

### CODE REVIEW

File: provider/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -66,8 +67,9 @@ func NewKubernetesProvider(
 }
 
 func (p *kubernetesProvider) GetFreeNodePort() (int, error) {
+	min, max := viper.GetInt("node-port-range-lower"), viper.GetInt("node-port-range-upper")
```
Comment (mrIncompetent) [inline provider/kubernetes/kubernetes.go]: Never evaluate flags outside of the cmd.

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: cmd/kubermatic-api/cmd/root.go [hunk 2]
```diff
@@ -117,6 +118,9 @@ func init() {
 	RootCmd.PersistentFlags().StringVar(&jwtKey, "jwt-key", "", "The JSON Web Token validation key, encoded in base64")
 	RootCmd.PersistentFlags().StringVar(&address, "address", ":8080", "The address to listen on")
 	RootCmd.PersistentFlags().StringVar(&masterKubeconfig, "master-kubeconfig", "", "When set it will overwrite the usage of the InClusterConfig")
+
+	RootCmd.PersistentFlags().IntVar(&nodePortLower, "node-port-range-lower", 12000, "Will readjust the lower bound of the NodePort range")
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: This should be a single parsed flag, with input matching `12000-12300` evaluating to `12000` as min and `12300` as max

File: cmd/kubermatic-api/cmd/root.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ import (
 
 var cfgFile, kubeConfig, dcFile, secretsFile, jwtKey, address, masterKubeconfig string
 var dev, auth bool
+var nodePortUpper, nodePortLower int
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: Please refer to these consistently as `min` and `max`

File: cmd/kubermatic-api/cmd/root.go (hunk 2 above)
Comment (scruplelesswizard) [reply cmd/kubermatic-api/cmd/root.go]: Flag should be called something like `client-api-server-port-range` as this is not defined or associated to the node ports assigned to the cluster

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]

**Pull request closed without merging**